AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -374,61 +375,67 @@ - (void)listWithPartition:(NSString *)partition
-                                  MSLogError([MSDataStore logTag], @"Not able to retrieve documents: %@",
-                                             [cosmosDbError localizedDescription]);
-                                  MSDataSourceError *dataSourceCosmosDbError = [[MSDataSourceError alloc] initWithError:cosmosDbError];
-                                  MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceCosmosDbError partition:partition documentType:documentType];
-                                  completionHandler(documents);
-                                  return;
-                                }
-
-                                // Deserialize the list payload and try to get the array of documents.
-                                NSError *deserializeError;
-                                id jsonPayload = [NSJSONSerialization JSONObjectWithData:(NSData *)data options:0 error:&deserializeError];
-                                if (!deserializeError && ![MSDocumentUtils isReferenceDictionaryWithKey:jsonPayload
-                                                                                                    key:kMSDocumentsKey
-                                                                                                keyType:[NSArray class]]) {
-                                  deserializeError =
-                                      [[NSError alloc] initWithDomain:kMSACDataStoreErrorDomain
-                                                                 code:MSACDataStoreErrorJSONSerializationFailed
-                                                             userInfo:@{NSLocalizedDescriptionKey : @"Can't deserialize documents"}];
-                                }
-                                if (deserializeError) {
-                                  MSDataSourceError *dataSourceDeserializeError =
-                                      [[MSDataSourceError alloc] initWithError:deserializeError];
-                                  MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceDeserializeError partition:partition documentType:documentType];
+      [self performCosmosDbOperationWithPartition:partition
+                                       documentId:nil
+                                       httpMethod:kMSHttpMethodGet
+                                         document:nil
+                                additionalHeaders:additionalHeaders
+                                additionalUrlPath:nil
+                                completionHandler:^(NSData *_Nullable data, NSHTTPURLResponse *_Nullable response,
+                                                    NSError *_Nullable cosmosDbError) {
+                                  // If not OK.
+                                  if (response.statusCode != MSACDocumentSucceededErrorCode) {
```

Comment:
Why do you use `MSACDocumentSucceededErrorCode` instead of `MSHTTPCodesNo200OK`?

Reply:
Seems it is already addressed.